app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,11 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def show
+    if not (@partner= Partner.find_by_id(params[:id]))
```

Comment:
Is this a good time to think about Null Objects?

Otherwise, I agree with @PurityControl: while this failure is somewhat more graceful than a simple app breakdown, it gives the user no actionable information.

Also, are we now really opposed to `unless`?